instrumentation/apache-camel-2.20/src/test/groovy/test/RestCamelTest.groovy

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package test
+
+import static io.opentelemetry.trace.Span.Kind.CLIENT
+import static io.opentelemetry.trace.Span.Kind.INTERNAL
+import static io.opentelemetry.trace.Span.Kind.SERVER
+
+import com.google.common.collect.ImmutableMap
+import io.opentelemetry.instrumentation.test.AgentTestRunner
+import io.opentelemetry.instrumentation.test.utils.PortUtils
+import io.opentelemetry.trace.attributes.SemanticAttributes
+import org.apache.camel.CamelContext
+import org.apache.camel.ProducerTemplate
+import org.springframework.boot.SpringApplication
+import org.springframework.context.ConfigurableApplicationContext
+import spock.lang.Shared
+
+class RestCamelTest extends AgentTestRunner {
+
+  @Shared
+  ConfigurableApplicationContext server
+  @Shared
+  int port
+
+  def setupSpec() {
+    withRetryOnAddressAlreadyInUse({
+      setupSpecUnderRetry()
+    })
+  }
+
+  def setupSpecUnderRetry() {
+    port = PortUtils.randomOpenPort()
+    def app = new SpringApplication(RestConfig)
+    app.setDefaultProperties(ImmutableMap.of("restServer.port", port))
+    server = app.run()
+    println getClass().name + " http server started at: http://localhost:$port/"
+  }
+
+  def cleanupSpec() {
+    if (server != null) {
+      server.close()
+      server = null
+    }
+  }
+
+  def "rest component - server and client call with jetty backend"() {
+    setup:
+    def camelContext = server.getBean(CamelContext)
+    ProducerTemplate template = camelContext.createProducerTemplate()
+
+    when:
+    // run client and server in separate threads to simulate "real" rest client/server call
+    new Thread(new Runnable() {
+      @Override
+      void run() {
+        template.sendBodyAndHeaders("direct:start", null, ImmutableMap.of("module", "firstModule", "unitId", "unitOne"))
+      }
+    }
+    ).start()
+
+    then:
+    assertTraces(1) {
+      trace(0, 5) {
+        it.span(0) {
+          name "start"
+          kind INTERNAL
+          attributes {
+            "camel.uri" "direct://start"
+          }
+        }
+        it.span(1) {
+          name "GET"
+          kind CLIENT
+          attributes {
+            "$SemanticAttributes.HTTP_METHOD.key" "GET"
+            "$SemanticAttributes.HTTP_STATUS_CODE.key" 200
+            "camel.uri" "rest://get:api/%7Bmodule%7D/unit/%7BunitId%7D"
+          }
+        }
+        it.span(2) {
+          name "/api/firstModule/unit/unitOne"
+          kind SERVER
```

Comment:
Would it be possible to make the method that generates parent span name protected and override it in rest decorator? Ideally the span name would be `/api/{module}/unit/{unitId}`

Reply:
Done, now also server span update for REST follows desired convention.